### PR TITLE
Use ENV for Branch Name

### PR DESCRIPTION
**Pull request by whyjustin**
Current Jenkinsfile uses scm.branch[0].name which gives `*/master` for master and the absolute branch name for feature branches. Use `env.BRANCH_NAME` which gives absolute branch name for all branches.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -137,7 +137,7 @@ node('ubuntu-zion') {
         archiveArtifacts artifacts: "${archiveName}.tar.gz", onlyIfSuccessful: true
       }
     }
-     if (branch != '*/master') {
+     if (branch != 'master') {
```
Comment (bigspotteddog) [inline Jenkinsfile]: Looks like a previous commit indented an extra space on this line.

File: Jenkinsfile [hunk 2]
```diff
@@ -32,7 +32,7 @@ node('ubuntu-zion') {
       OsTools.runSafe(this, "docker system prune -a -f")
 
       checkout scm
-      branch = scm.branches[0].name
+      branch = env.BRANCH_NAME
```
Comment (bigspotteddog) [inline Jenkinsfile]: This evaluates to null for me.

Please see: https://jenkins.zion.aws.s/job/integrations/job/jcava-docker-nexus3/1/console

[Pipeline] echo
branch = null

Comment (whyjustin) [reply Jenkinsfile]: Thanks, looks like that's only set in multibranch pipeline jobs. 
11cd3c3

File: Jenkinsfile [hunk 3]
```diff
@@ -31,12 +31,12 @@ node('ubuntu-zion') {
...
       OsTools.runSafe(this, "docker system prune -a -f")
 
-      checkout scm
+      def checkoutDetails = checkout scm
 
       dockerFileLocation = "${pwd()}/Dockerfile"
 
-      branch = OsTools.runSafe(this, 'git rev-parse --abbrev-ref HEAD')
-      commitId = OsTools.runSafe(this, 'git rev-parse HEAD')
+      branch = checkoutDetails.GIT_BRANCH
```
Comment (bigspotteddog) [inline Jenkinsfile]: branch now equals 'origin/master' which is good. The master branch check is checking for 'master'.

Comment (bigspotteddog) [reply Jenkinsfile]: I like the checkoutDetails object 👍

Comment (whyjustin) [reply Jenkinsfile]: Normalized it like [0]. AFAICT there's no way to always get the local branch name and this is required to checkout attached to update version and sha.

[0] https://github.com/sonatype/docker-nexus3/pull/94/commits/633de316c938321517faa649e8a68cf1355f1d5c#diff-58231b16fdee45a03a4ee3cf94a9f2c3R38

**Pull request merged**